### PR TITLE
YLP-2151 As a HCP I want to be able to filter my patients via the new…

### PR DESCRIPTION
**Pull request by TimCanu**
… design

## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [x] End-to-end tests successful on CI

### CODE REVIEW

File: packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx
Comment (fatb38) [inline packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:1]: **👏 praise:** ‏nice test !

File: packages/yourloops/test/integration/pages/profile/caregiver-profile.spec.tsx [hunk 1]
```diff
@@ -51,7 +51,7 @@ import { mockUserApi } from '../../mock/user.api.mock'
 import { mockAuthApi } from '../../mock/auth.api.mock'
 import { Unit } from 'medical-domain'
 
-describe('Caregiver page for hcp', () => {
+describe('Profile page for caregiver', () => {
```
Comment (fatb38) [inline packages/yourloops/test/integration/pages/profile/caregiver-profile.spec.tsx:54]: sharp eye 😜

File: packages/yourloops/lib/patient/patient.hook.tsx [hunk 2]
```diff
@@ -83,38 +83,35 @@ export default function usePatientProviderCustomHook(): PatientContextResult {
     setPatients(patientsUpdated)
   }, [patients])
 
-  const patientsForSelectedTeam = user.isUserHcp() ? patients.filter((patient: Patient) => patient.teams.some((team: PatientTeam) => team.teamId === selectedTeam.id)) : []
-  const patientList = user.isUserHcp() ? patientsForSelectedTeam : patients
+  const patientsForSelectedTeam = useMemo(() => {
+    if (!user.isUserHcp()) {
+      return patients
+    }
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/lib/patient/patient.hook.tsx:89]: I don't understand this comment and I don't think that this is useless :thinking:

File: packages/yourloops/test/integration/data/patient.api.data.ts [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+export const timeSpentOutOfTargetRangePatientId = 'timeSpentOutOfTargetRangePatientId'
+export const hypoglycemiaPatientId = 'hypoglycemiaPatientId'
+export const noDataTransferredPatientId = 'noDataTransferredPatientId'
+export const flaggedPatientId = 'flaggedPatientId'
+
+const defaultMonitoring: Monitoring = {
+  enabled: true,
+  monitoringEnd: getTomorrowDate(),
+  status: MonitoringStatus.accepted,
+  parameters: monitoringParameters
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/test/integration/data/patient.api.data.ts:63]: This was here before and I don't know how to fix it. :wink:

File: packages/yourloops/test/integration/data/patient.api.data.ts [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+export const monitoredPatientWithMmol: Patient = buildPatient(
+  monitoredPatientWithMmolId,
+  [
+    {
+      teamId: myThirdTeamId,
+      status: UserInvitationStatus.accepted,
+      monitoringStatus: MonitoringStatus.accepted
+    }
+  ],
+  { ...defaultMonitoring, parameters: monitoringParametersBgUnitMmol },
```
[the remark this answers is not in the document]
Comment (TimCanu) [reply packages/yourloops/test/integration/data/patient.api.data.ts:326]: Same

File: packages/yourloops/lib/patient/patient.util.ts [hunk 5]
```diff
@@ -102,38 +106,49 @@ export default class PatientUtils {
...
-        return patients.filter(patient => patient.monitoring?.enabled && patient.monitoring.monitoringEnd && new Date(patient.monitoring.monitoringEnd).getTime() - twoWeeksFromNow.getTime() < 0)
-      default:
-        return patients
+  static extractPatients = (patients: Patient[], patientFilters: PatientsFilters, flaggedPatientsId: string[] | undefined, selectedTeamId: string): Patient[] => {
+    // When the filter is pending, we only get the pending patients and don't apply any filter on them
+    if (patientFilters.pendingEnabled) {
+      return patients.filter((patient) => PatientUtils.isInvitationPending(patient, selectedTeamId))
+    }
+    // We do not take the pending patients
+    let patientsExtracted = PatientUtils.getNonPendingPatients(patients, selectedTeamId)
```
Comment (celia-rouis) [inline packages/yourloops/lib/patient/patient.util.ts:140]: **🎯 suggestion (non-blocking):** ‏I found another way of doing the conditional filtering that seems shorter and avoids the reassignment via `let`, tell me what you think!

```ts
const nonPendingPatients = PatientUtils.getNonPendingPatients(patients, selectedTeamId)
    
return PatientUtils.filterPatientsOnMonitoringAlerts(nonPendingPatients, patientFilters)
  .filter(patient => patientFilters.telemonitoredEnabled ? patient.monitoring?.enabled : patient)
  .filter(patient => patientFilters.manualFlagEnabled ? flaggedPatientsId?.includes(patient.userid) : patient)
  .filter(patient => patientFilters.messagesEnabled ? patient.metadata.hasSentUnreadMessages : patient)
```

**Review by fatb38 — APPROVED** [review]

**Pull request merged**